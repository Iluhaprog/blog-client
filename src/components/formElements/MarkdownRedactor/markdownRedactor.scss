@import '../../../vars.scss';
@import '~github-markdown-css/github-markdown.css';

.markdown-redactor {
    width: 100%;

    &_big {
        display: flex;
        flex-direction: row;
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        height: 100vh;
        background-color: $text-color-2;
        z-index: 2;
        
        .markdown-redactor__view-zone {
            display: block;
            padding: 20px;
            width: 50%;
            height: 100%;
            overflow-y: scroll;
            background-color: #fff;
        }

        .markdown-redactor__write-zone {
            width: 50%;
            height: 100%;
            border-right: $border-1;
            border-width: 5px;
            textarea {
                height: 100%;
                border-radius: 0;
            }
        }
    }

    &__write-zone {
        position: relative;
        .button {
            position: absolute;
            left: 0;
            font-size: 14px !important;
            opacity: 0;
            transition: .2s ease-in-out;
        }
        &:hover {
            .button {
                left: calc(100% - 75px);
                opacity: 1;
            }
        }
    }

    &__view-zone {
        display: none;
    }
}