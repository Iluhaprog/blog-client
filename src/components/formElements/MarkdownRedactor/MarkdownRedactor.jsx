import React, { useState } from 'react';
import marked from 'marked';
import { Textarea } from '../Textarea';
import './markdownRedactor.scss';
import { SuccessButton, DangerButton } from '../../buttons';

export default props => {
    const [view, setView] = useState(false);
    const html = marked(props.value || '');
    return (
        <div className={`markdown-redactor ${view ? 'markdown-redactor_big' : ''}`}>
            <div className='markdown-redactor__write-zone'>
                { view 
                    ? (
                        <DangerButton 
                            text='close'
                            onClick={() => setView(!view)}
                        />
                      )
                    : (
                        <SuccessButton 
                            text='view'
                            onClick={() => setView(!view)}
                        />
                    )
                }
                <Textarea 
                    name={props.name}
                    placeholder={props.placeholder}
                />
            </div>
            <div 
                className='markdown-redactor__view-zone markdown-body'
                dangerouslySetInnerHTML={{__html: html}}
            >
            </div>
        </div>
    )
}