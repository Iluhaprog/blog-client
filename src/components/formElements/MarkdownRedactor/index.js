import MarkdownRedactor from './MarkdownRedactor';

export {
    MarkdownRedactor,
};