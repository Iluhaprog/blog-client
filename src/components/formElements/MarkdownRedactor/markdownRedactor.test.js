import React from 'react';
import configureMockStore from 'redux-mock-store';
import { reduxForm } from "redux-form";
import { Provider } from 'react-redux';
import { render, fireEvent, waitFor } from '@testing-library/react';
import MarkdownRedactor from './MarkdownRedactor';


const mockStore = configureMockStore();
const decor = el => {
    const form = () => (<div>{el}</div>);
    return reduxForm({
        form: 'test',
    })(form);
}

test('Test view of markdown redactor', () => {
    const store = mockStore();
    const Mr = decor(
        <MarkdownRedactor 
            name='text'
            placeholder='Text'
            value={''}
        />
    );
    const { asFragment } = render(
        <Provider store={store}>
            <Mr />
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

test('Test view of markdown redactor full screen', async () => {
    const store = mockStore();
    const Mr = decor(
        <MarkdownRedactor 
            name='text'
            placeholder='Text'
            value={''}
        />
    );
    const { asFragment, container } = render(
        <Provider store={store}>
            <Mr />
        </Provider>
    );
    fireEvent.click(container.querySelector('.success-button'));
    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    })
});