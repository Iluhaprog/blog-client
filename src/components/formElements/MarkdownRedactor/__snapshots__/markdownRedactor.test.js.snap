// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view of markdown redactor 1`] = `
<DocumentFragment>
  <div>
    <div
      class="markdown-redactor "
    >
      <div
        class="markdown-redactor__write-zone"
      >
        <button
          class="button default-button success-button"
        >
          view
        </button>
        <textarea
          class="textarea"
          name="text"
          placeholder="Text"
        />
      </div>
      <div
        class="markdown-redactor__view-zone markdown-body"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test view of markdown redactor full screen 1`] = `
<DocumentFragment>
  <div>
    <div
      class="markdown-redactor markdown-redactor_big"
    >
      <div
        class="markdown-redactor__write-zone"
      >
        <button
          class="button default-button danger-button"
        >
          close
        </button>
        <textarea
          class="textarea"
          name="text"
          placeholder="Text"
        />
      </div>
      <div
        class="markdown-redactor__view-zone markdown-body"
      />
    </div>
  </div>
</DocumentFragment>
`;
